ini/parse.ts

```diff
@@ -1,9 +1,63 @@
 // Copyright 2018-2025 the Deno authors. MIT license.
 // This module is browser compatible.
 
-import { IniMap, type ReviverFunction } from "./_ini_map.ts";
+import type { ReviverFunction } from "./_ini_map.ts";
 export type { ReviverFunction };
 
+const SECTION_REGEXP = /^\[(?<name>\S.*\S|\S+)]$/;
```

Comment:
Now this doesn't match with `" a "`, for example. `parse("[ a ]\na=1")` returns `{ " a ": { a: 1 } }` on main. It throws in this branch.

I'm not sure the existing check can be done as a single regexp check.